httptransport/server.go

```diff
@@ -20,12 +20,12 @@ import (
 
 const (
 	apiRoot                 = "/api/v1/"
-	VulnerabilityReportPath = apiRoot + "vulnerability_report/"
-	IndexAPIPath            = apiRoot + "index_report"
-	IndexReportAPIPath      = apiRoot + "index_report/"
-	StateAPIPath            = apiRoot + "state"
+	VulnerabilityReportPath = apiRoot + "matcher/vulnerability_report/"
```

Comment:
I understand why you are doing this, but keep these things in mind:
- The API implementation is bending to the limitations of OpenShift router
- The API resources are now explicitly tied to different services under the prefix
- The API is moving further away from being RESTful, which may not matter, but helps clair-ify (pun intended) future API decisions

When there is full service discovery, the paths don't _actually_ matter, because the clients don't have to hardcode them. What they care about are _operation IDs_ (like `GetIndexReport`), which if they can be discovered from a single OpenAPI endpoint, solves all your problems and you are free to route your services however you want. That seems like a much more scalable and supportable long-term solution, which because the API is small right now, is feasible without a lot of effort. And I would be happy to help.

Reply:
"The API implementation is bending to the limitations of OpenShift router"

That is not true. This change makes placing any path router in front of Clair easier. Think about what an ops person needs to do. They can simply route "/indexer/" prefix to the backend service with this change and little configuration.

The alternative is having the ops person configure an arbitrary prefix, having their load balancer strip that prefix, and sending the stripped suffix to the services. Their clients  will need to change, and in a way dependent on their infra when deployed on prem. THIS example requires clients talking to the path router to bend to the infra. 

The other, other, alternative is specifying a path route per api endpoint in a path router config. This is not great either as every time the services add an endpoint the ops engineer needs to update their infra, they must provide larger configs to their path router, and the margin for error increases meaning more tickets. 

"The API resources are now explicitly tied to different services under the prefix"

But... this is how we "do" micro-services.? Each service owns a resource, each resource has a route which points to particular service. What goals do you accomplish by decoupling http routes from their services? Do we expect index_reports to be suddenly handles by the matcher service?

"The API is moving further away from being RESTful, which may not matter, but helps clair-ify (pun intended) future API decisions"

I dont see how adding a "root" is breaking RESTFUL. Just as "api/v1" is a common root. "api/v1/indexer" is a common route for indexer resources. 


Also, i havent seen OpenApi used as a mechanism to route container and server traffic before. I am under the impression most ppl will be using ingress path routing, nginx, or traefik.